lib/matplotlib/axes/_axes.py

```diff
@@ -6142,6 +6147,16 @@ def _normalize_input(inp, ename='input'):
                 raise ValueError(
                     'weights should have the same shape as x')
 
+        # Combine the masks from x[i] and w[i] (if applicable) into a single
+        # mask and apply it to both.
+        if not input_empty:
+            for i in range(len(x)):
+                mask_i = x[i].mask
+                if w[i] is not None:
+                    mask_i = mask_i | w[i].mask
```

Comment:
But that explains the mask on x, not on the weights. I am also fine with a mask on the weights as long as it is identical as the one on x. But masks different on the weights and on x seem like it may be an error on the users side, hence throwing a ValueError.